downloader/src/main/kotlin/Downloader.kt

```diff
@@ -56,7 +59,7 @@ import org.ossreviewtoolkit.utils.withoutPrefix
 /**
  * The class to download source code. The signatures of public functions in this class define the library API.
  */
-object Downloader {
+class Downloader(private val config: DownloaderConfiguration = DownloaderConfiguration()) {
     private fun verifyOutputDirectory(outputDirectory: File) {
         require(!outputDirectory.exists() || outputDirectory.list().isEmpty()) {
```

Comment:
Proposal: Should the Downloader check that the list of sources in the configuration is not empty as otherwise no downloads are possible? Or should the configuration itself already check this?

Reply:
I added an init function to the DownloaderConfiguration to check if the configuration is valid/sane.